### ISSUE TITLE
SideNav does not collapse when it should

### ISSUE THREAD

**Issue by nxn-4-wdf** [carbon-design-system/carbon#3710]
## What package(s) are you using?

- [x] `carbon-components`
- [x] `carbon-components-react`

## Detailed description

> Describe in detail the issue you're having.

This is a corner case: on a large viewport, the SideNav is always collapsed per default, but with a combination of resizing and actions, you can have it displayed it anyway.

> What did you expect to happen? What happened instead? What would you like to
> see changed?

SideNav shall always be hidden per default on large viewports, even if it is in state expanded or hover.
```SCSS
  .#{$prefix}--side-nav:not(.#{$prefix}--side-nav--fixed):hover,
  .#{$prefix}--side-nav--expanded {
    width: mini-units(32);
  }
``` 

> What version of the Carbon Design System are you using?

10.4

## Steps to reproduce the issue

Example app on: https://jtrzb.csb.app/#/

Test with `expanded`:
1. Reduce the browser window (viewport) width so that the HeaderMenuButton displays on the top-left corner:
![image](https://user-images.githubusercontent.com/52698706/62852179-f7f56580-bce8-11e9-9be0-fcc4a623aecd.png)
2. Click on the HeaderMenuButton , the SideNav opens:
![image](https://user-images.githubusercontent.com/52698706/62852217-15c2ca80-bce9-11e9-9c5a-e0eba0d88c7d.png)
3. Enlarge the window width again. The HeaderMenuButton and the overlay collapse, but the SideNav remains, and hides the content:
![image](https://user-images.githubusercontent.com/52698706/62852282-46a2ff80-bce9-11e9-8a63-d610b39e12cf.png)

Test with `:hover`:
1. Same as 1. above
2. Same as 2. above. Notice that the current active Element is the HeaderMenuButton, which now displays as a cross
3. Put the mouse cursor (not displayed on the screenshot below) over SideNav link "One":
![image](https://user-images.githubusercontent.com/52698706/62853263-1315a480-bcec-11e9-8b29-810468910755.png)
4. Without moving the mouse cursor, press the Enter-key, which "clicks" on the cross:
![image](https://user-images.githubusercontent.com/52698706/62853381-769fd200-bcec-11e9-96b6-ca6af8d03c8a.png)

Responsible seems to be in _side_nav.scss line 113:

> Please create a reduced test case in CodeSandbox

Cose in CodeSandbox: https://codesandbox.io/s/sidenavissues-jtrzb

## Additional information

- Screenshots or code: see above.

**Comment by jnm2377** [comment]
Hi 👋 thanks for filing this issue. It appears that the first issue is not a bug. By default, our sidenav is persistent at large screensizes, always. You can tell by the fact that the menu button also disappears. There is a prop that allows this to be false, but the actual functionality for allowing it to be toggle-able at all sizes hasn't been developed yet. So it is expected that the sidenav will show up at the larger sizes, at least until we develop a variation that doesn't. 

It does appear like the second item is an issue, however. Thanks for catching that!

**Comment by jmmadsen** [comment]
@jnm2377 
`There is a prop that allows this to be false, but the actual functionality for allowing it to be toggle-able at all sizes hasn't been developed yet. So it is expected that the sidenav will show up at the larger sizes, at least until we develop a variation that doesn't.`

Has this been updated yet? I have a use case where I would always like the SideNav to be toggleable, at all screen sizes.

**Comment by Erichx** [comment]
@jmmadsen  Hello!
I was looking for the same solution as you, and I found a little hack.
Use this example to your project [click here](https://codesandbox.io/s/header-with-sidenav-1u7su?from-embed=&file=/src/index.js:5679-5686)
So, then go to your css file and add this two simple classes:
`.bx--side-nav--ux{ width: 0px;}
nav.bx--side-nav--expanded.bx--side-nav--ux  {
  width: 16rem;
}`

Works fine on my project, ping me if something went weird.
Hope I helped you ;)
